src/types.ts

```diff
@@ -61,61 +64,51 @@ export interface SafeInfo {
   ethBalance: string;
 }
 
-export interface TxConfirmationEvent {
-  requestId: RequestId;
-  safeTxHash: string;
-}
+export type Methods = keyof typeof METHODS;
 
-export interface TxRejectionEvent {
-  requestId: RequestId;
-}
+export type SDKRequestData<M extends Methods = Methods, P = unknown> = {
+  id: RequestId;
+  params: P;
+  env: {
+    sdkVersion: string;
+  };
+  method: M;
+};
 
-export type InterfaceMessageIds = keyof typeof INTERFACE_MESSAGES;
+export type SDKMessageEvent = MessageEvent<SDKRequestData>;
 
-export interface InterfaceMessageEvent extends MessageEvent {
-  data: {
-    requestId: RequestId;
-    messageId: InterfaceMessageIds;
-    data: InterfaceMessageToPayload[InterfaceMessageIds];
-  };
-}
+export type ErrorResponse = {
+  id: RequestId;
+  success: false;
+  error: string;
+  version?: string;
+};
 
-export interface SDKMessageToPayload {
-  [SDK_MESSAGES.SAFE_APP_SDK_INITIALIZED]: undefined;
-  [SDK_MESSAGES.SEND_TRANSACTIONS]: Transaction[];
-  [SDK_MESSAGES.RPC_CALL]: {
-    call: RpcCallNames;
-    params: unknown[];
-  };
-  [SDK_MESSAGES.SEND_TRANSACTIONS_V2]: {
-    txs: Transaction[];
-    params?: SendTransactionParams;
-  };
-}
+export type SuccessResponse<T = MethodToResponse[Methods]> = {
+  id: RequestId;
+  data: T;
+  version?: string;
+  success: true;
+};
 
-export type SDKMessageIds = keyof typeof SDK_MESSAGES;
+export type Response<T = MethodToResponse[Methods]> = ErrorResponse | SuccessResponse<T>;
 
-export interface InterfaceMessageToPayload {
-  [INTERFACE_MESSAGES.ON_SAFE_INFO]: SafeInfo;
-  [INTERFACE_MESSAGES.TRANSACTION_CONFIRMED]: {
-    safeTxHash: string;
-  };
-  [INTERFACE_MESSAGES.ENV_INFO]: {
-    txServiceUrl: string;
-  };
-  [INTERFACE_MESSAGES.TRANSACTION_REJECTED]: Record<string, unknown>;
-  [INTERFACE_MESSAGES.RPC_CALL_RESPONSE]: unknown;
-}
+export type InterfaceMessageEvent = MessageEvent<Response>;
 
-export type RPCPayload<R extends RpcCallNames, P extends unknown> = {
-  call: R;
-  params: P;
+export type EnvInfo = {
+  txServiceUrl: string;
 };
 
-export type SentSDKMessage<T extends SDKMessageIds, D = SDKMessageToPayload[T]> = {
-  messageId: T;
-  requestId: RequestId;
-  data: D;
+export interface MethodToResponse {
```

Comment:
Just out of curiousity: why is there no mapping for MethodToRequest?

Reply:
To be honest the types turned me crazy a little bit... Mappings don't really work well and I couldn't find a better way, so I left it for future improvements

Reply:
Makes sense :)